### PR TITLE
[FLINK-25959] Add micro-benchmarks for the sort-based blocking shuffle

### PR DESCRIPTION
**Pull request by wsry**
This patch adds micro-benchmarks for the sort-based blocking shuffle.

### CODE REVIEW

**Comment by wsry** [comment]
@pnowojski As discussed in FLINK-25704, I add a micro-benchmark for the sort-based blocking shuffle, please take a look when you have time.

File: src/main/java/org/apache/flink/benchmark/BlockingPartitionBenchmark.java [hunk 1]
```diff
@@ -66,6 +66,16 @@ public void uncompressedMmapPartition(UncompressedMmapEnvironmentContext context
...
+    @Benchmark
+    public void compressedSortShufflePartition(CompressedSortShuffleEnvironmentContext context) throws Exception {
+        executeBenchmark(context.env);
+    }
+
+    @Benchmark
+    public void uncompressedSortShufflePartition(UncompressedSortShuffleEnvironmentContext context) throws Exception {
+        executeBenchmark(context.env);
+    }
+
```
Comment (pnowojski) [inline src/main/java/org/apache/flink/benchmark/BlockingPartitionBenchmark.java:78]: nit: rename to `compressedSortPartition` and `uncompressedSortPartition` to shorten the benchmark name in the web UI?

File: src/main/java/org/apache/flink/benchmark/BlockingPartitionRemoteChannelBenchmark.java [hunk 2]
```diff
@@ -49,7 +49,15 @@ public static void main(String[] args) throws RunnerException {
...
-    public void remoteFilePartition(BlockingPartitionEnvironmentContext context) throws Exception {
+    public void remoteFilePartition(RemoteFileEnvironmentContext context) throws Exception {
+        StreamGraph streamGraph =
+                StreamGraphUtils.buildGraphForBatchJob(context.env, RECORDS_PER_INVOCATION);
+        context.miniCluster.executeJobBlocking(
+                StreamingJobGraphGenerator.createJobGraph(streamGraph));
+    }
+
+    @Benchmark
+    public void remoteSortShufflePartition(RemoteSortShuffleEnvironmentContext context) throws Exception {
```
Comment (pnowojski) [inline src/main/java/org/apache/flink/benchmark/BlockingPartitionRemoteChannelBenchmark.java:60]: nit ditto: `remoteSortPartition`?

**Review by pnowojski — APPROVED** [review]

**Comment by wsry** [comment]
Merging.

**Pull request merged**